wincode-derive/src/schema_read.rs

```diff
@@ -225,27 +230,249 @@ fn impl_struct_extensions(args: &SchemaArgs) -> Result<TokenStream> {
+                /// to guarantee that the `MaybeUninit<T>` really is in an initialized state.
+                #[inline]
+                #vis const unsafe fn into_assume_init_mut(mut self) -> &'_wincode_inner mut #builder_dst {
+                    // SAFETY: reference lives beyond the scope of the builder, and builder is forgotten.
+                    let inner = unsafe { ptr::read(&mut self.inner) };
+                    mem::forget(self);
+                    // SAFETY: Caller asserts the `MaybeUninit<T>` is in an initialized state.
+                    unsafe {
+                        inner.assume_init_mut()
+                    }
+                }
+
+                /// Forget the builder, disabling the drop logic.
+                ///
+                /// # Safety
+                ///
+                /// Assuming the source `MaybeUninit<T>` when the content is not yet fully initialized causes undefined behavior.
+                /// It is up to the caller to guarantee that the `MaybeUninit<T>` really is in an initialized state.
+                #[inline]
+                #vis const unsafe fn assume_init_forget(self) {
+                    mem::forget(self);
+                }
+            }
+        }
+    };
+
+    // Generate the helper methods for the builder.
+    let builder_helpers = fields.iter().enumerate().map(|(i, field)| {
+        let target = field.target_resolved();
+        let target_reader_bound = target.with_lifetime("de");
+        let ident = field.struct_member_ident(i);
+        let ident_string = field.struct_member_ident_to_string(i);
+        let uninit_mut_ident = format_ident!("uninit_{ident_string}_mut");
+        let read_field_ident = format_ident!("read_{ident_string}");
+        let write_uninit_field_ident = format_ident!("write_{ident_string}");
+        let assume_init_field_ident = format_ident!("assume_init_{ident_string}");
+        let init_with_field_ident = format_ident!("init_{ident_string}_with");
+
+        // The bit index for the field.
+        let index_bit = LitInt::new(&(1 << i).to_string(), Span::call_site());
+        let set_index_bit = quote! {
+            self.init_set |= #index_bit;
+        };
+
+        quote! {
+            /// Get a mutable reference to the maybe uninitialized field.
+            #[inline]
+            #vis const fn #uninit_mut_ident(&mut self) -> &mut MaybeUninit<#target> {
+                // SAFETY:
+                // - `self.inner` is a valid reference to a `MaybeUninit<#builder_dst>`.
+                // - We return the field as `&mut MaybeUninit<#target>`, so
+                //   the field is never exposed as initialized.
+                unsafe { &mut *(&raw mut (*self.inner.as_mut_ptr()).#ident).cast() }
+            }
+
+            /// Write a value to the maybe uninitialized field.
+            #[inline]
+            #vis const fn #write_uninit_field_ident(&mut self, val: #target) -> &mut Self {
```

Comment:
I think the choice of `read_field_name` and `write_field_name` names is a bit confusing here (though I guess it's already inherited from previous design of `struct_extensions`), since `read` operates on `Reader`, while `write` doesn't have anything to do with wincode io `Writer`.

I would definitely prefer to change `write` into something like `store_field_name` or `init_field_name`, which would also convey the message that field is not manually initialized